app/pages/gsoc/gsoc2022.js

```diff
@@ -0,0 +1,91 @@
+import Head from "next/head";
+import { fetchAPI } from "../../lib/api";
+import { Container, Col } from "react-bootstrap";
+import styles from "../../styles/Leaderboard.module.css";
+import LeaderboardTable from "../../components/leaderBoardTable";
+
+export default function LeaderBoard({ contributors }) {
+  return (
+    <div>
+      <Head>
+        <title>GSOC2022 LeaderBoard</title>
+        <meta
+          name="description"
+          content="Rocket.Chat LeaderBoard for GSOC2022"
+        />
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+      <Container
+        fluid
+        className="d-flex flex-column align-items-center gap-2 gap-md-5"
+      >
+        <Col className="d-flex flex-column align-items-center gap-2 pt-3">
+          <h1
+            className={`display-6 fw-bold text-center ${styles.hero_heading}`}
+          >
+            <span className={styles.redText}>Rocket.Chat</span> GSoC
+            Contribution Leaderboard
+          </h1>
+          <p
+            className={`fw-regular col-10 col-md-8 text-center ${styles.hero_subheading}`}
+          >
+            Start your open-source jourey with rocket.chat
+          </p>
+        </Col>
+        <Col className="d-flex flex-column align-items-center col-10 col-md-8">
+          <LeaderboardTable contributors={contributors} />
+        </Col>
+      </Container>
+    </div>
+  );
+}
+
+export async function getServerSideProps() {
+  const res = await fetch("https://gsoc.rocket.chat/api/data");
```

Comment:
No hard-coding of URL for re-usable components.   Remember that other community builders will be going to their own (not Rocket.Chat's)  leaderboard.

Reply:
> Great work @samad-yar-khan ! 👏
> 
> Please add argument for TopN where say if N=10, then the component will change layout and only display the Top 10 ranked in a 1 bootstrap column by 3 bootsrap row format. Only display -- heading, ranking, avatar, and username. (no need for merge request counts etc)
> 
> It would be good if the width of the browser is shrunk (or when using mobile browser) that this component will reactively shrink down to a similar format.

Thanks a lot, I will add the feature to display top N, and reducing the number of columns for smaller screens. I will rename the css files as well.